### PR TITLE
install specific version of protoc-gen-go

### PR DESCRIPTION
**Pull request by schmichael**
Lots of build updates to try to make them more reproducible:
* Fix protoc install (thanks @notnoop!)
* Install specific version of protoc-gen-go
* Install specific version of codec/codecgen
* Delete generated files in `make clean`

### CODE REVIEW

**Comment by schmichael** [comment]
Actually this didn't work. Got this when running `make prerelease` in a fresh Vagrant box:

```
vagrant@linux:/opt/gopath/src/github.com/hashicorp/nomad$ make prerelease



google/protobuf/timestamp.proto: Read access is denied for file: /usr/local/include/google/protobuf/timestamp.proto
google/protobuf/duration.proto: Read access is denied for file: /usr/local/include/google/protobuf/duration.proto
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto: Import "google/protobuf/duration.proto" was not found or had errors.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto: Import "google/protobuf/timestamp.proto" was not found or had errors.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:193:5: "google.protobuf.Duration" is not defined.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:236:5: "google.protobuf.Duration" is not defined.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:268:5: "google.protobuf.Duration" is not defined.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:503:5: "google.protobuf.Timestamp" is not defined.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:507:5: "google.protobuf.Timestamp" is not defined.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:526:5: "google.protobuf.Timestamp" is not defined.
github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto:597:5: "google.protobuf.Timestamp" is not defined.
drivers/shared/executor/proto/executor.proto: Import "google/protobuf/timestamp.proto" was not found or had errors.
drivers/shared/executor/proto/executor.proto: Import "github.com/hashicorp/nomad/plugins/drivers/proto/driver.proto" was not found or had errors.
drivers/shared/executor/proto/executor.proto:22:5: "hashicorp.nomad.plugins.drivers.proto.Resources" is not defined.
...
```

Any ideas @nickethier ?

**Comment by notnoop** [comment]
Looks like there was a permission problem with protoc binary installation - that's not related to this change.  I wonder if generation ever worked on Vagrant - the following diff makes it work

```diff
diff --git a/scripts/vagrant-linux-priv-protoc.sh b/scripts/vagrant-linux-priv-protoc.sh
index 184fa049c..d2e3ae853 100755
--- a/scripts/vagrant-linux-priv-protoc.sh
+++ b/scripts/vagrant-linux-priv-protoc.sh
@@ -19,14 +19,19 @@ function install_protoc() {
     # Unzip
     unzip /tmp/protoc.zip -d /tmp/protoc3

+    # all protoc files should be world-wide readable, specially the include files
+    chmod -R a+r /tmp/protoc3
+
     # Move protoc to /usr/local/bin/
-    sudo mv /tmp/protoc3/bin/* /usr/local/bin/
+    mv /tmp/protoc3/bin/* /usr/local/bin/

     # Move protoc3/include to /usr/local/include/
-    sudo mv /tmp/protoc3/include/* /usr/local/include/
+    mv /tmp/protoc3/include/* /usr/local/include/

     # Link
-    sudo ln -s /usr/local/bin/protoc /usr/bin/protoc
+    ln -s /usr/local/bin/protoc /usr/bin/protoc
+
+    rm -rf /tmp/protoc3 /tmp/protoc.zip
 }

 install_protoc
```

**Comment by schmichael** [comment]
Thanks @notnoop! Pushed that fix as well as a few more changes to get our builds closer to reproducible. See PR description for details.

File: GNUmakefile [hunk 1]
```diff
@@ -280,6 +282,7 @@ clean: ## Remove build artifacts
 	@rm -rf "$(PROJECT_ROOT)/bin/"
 	@rm -rf "$(PROJECT_ROOT)/pkg/"
 	@rm -f "$(GOPATH)/bin/nomad"
+	@find client nomad -name '*.generated.*' -delete
```
Comment (notnoop) [inline GNUmakefile]: That hit few times - thanks for addressing it!

**Pull request merged**